WeatherApp/DAL/WeatherRepository.cs

```diff
@@ -26,5 +31,29 @@ public async Task<double> GetTemperatureByCityNameAsync(string cityName)
 
             return weather.Main.Temp;
         }
+
+        public async Task<IEnumerable<WeatherForecast>> GetForecastByCoordinatesAsync(Coordinates coordinates, DateTime startDate, DateTime endDate)
+        {
+            using var response = await _httpClient.GetAsync($"https://api.open-meteo.com/v1/forecast?latitude={coordinates.Latitude.ToString(CultureInfo.InvariantCulture)}&longitude={coordinates.Longitude.ToString(CultureInfo.InvariantCulture)}&daily=temperature_2m_max,temperature_2m_min&timezone=auto&start_date={startDate.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)}&end_date={endDate.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)}");
```

Comment:
The same issue with HttpClient. Please create client under "using" construction

Reply:
HttpClient is now created using IHttpClientFactory and disposed via "using".